openshift-kube-apiserver/openshiftkubeapiserver/patch.go

```diff
@@ -109,7 +110,26 @@ func OpenShiftKubeAPIServerConfigPatch(genericConfig *genericapiserver.Config, k
 	// END HANDLER CHAIN
 
 	openshiftAPIServiceReachabilityCheck := newOpenshiftAPIServiceReachabilityCheck(genericConfig.PublicAddress)
-	oauthAPIServiceReachabilityCheck := newOAuthPIServiceReachabilityCheck(genericConfig.PublicAddress)
+
+	oauthAPIServiceTerminationCondition := func() (bool, string) {
+		authnCache := authncache.NewAuthnCache(openshiftInformers.OpenshiftConfigInformers.Config().V1().Authentications())
+		err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+			return authnCache.HasSynced(), nil
+		})
+		if err == nil {
+			auth, err := authnCache.Authn()
+			if err == nil && auth != nil {
+				if auth.Spec.Type == configv1.AuthenticationTypeOIDC {
```

Comment:
Probably want something a bit more robust than just checking this. May be able to have this check the configuration to see if it should be running in OIDC mode.